### PR TITLE
Log put counts in CloudBlobStore for debugging.

### PR DESCRIPTION
**Pull request by lightningrob**
Better debug messages for token persistence.

### CODE REVIEW

File: ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java [hunk 1]
```diff
@@ -160,10 +168,26 @@ private void putBlob(MessageInfo messageInfo, ByteBuffer messageBuf, long size)
...
       logger.trace("Blob is skipped: {}", messageInfo);
       vcrMetrics.blobUploadSkippedCount.inc();
+      numPutsSkipped++;
+    }
+  }
+
+  /**
+   * Log the collected put counts on the prescribed interval.
+   */
+  private void logPutCountsPeriodically() {
```
Comment (zzmao) [inline ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java]: Why not use a Guage?  Guage can dispaly in-memory variables. 

Actually, we have all these vars in azureMetrics
```
  public final Counter blobUploadRequestCount;
  public final Counter blobUploadSuccessCount;
  public final Counter blobUploadConflictCount;
```
We can have a Guage version of them.

File: ambry-replication/src/main/java/com.github.ambry.replication/ReplicaTokenPersistor.java [hunk 2]
```diff
@@ -41,7 +41,7 @@
  */
 public abstract class ReplicaTokenPersistor implements Runnable {
 
-  private static final Logger logger = LoggerFactory.getLogger(DiskTokenPersistor.class);
+  private static final Logger logger = LoggerFactory.getLogger(ReplicaTokenPersistor.class);
```
Comment (jsjtzyy) [inline ambry-replication/src/main/java/com.github.ambry.replication/ReplicaTokenPersistor.java]: Thanks for catching this.

File: ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java [hunk 3]
```diff
@@ -64,6 +64,10 @@
   private final long minTtlMillis;
   private final boolean requireEncryption;
   private boolean started;
+  // Internal counts for put requests to this partition store since store startup (includes all replicas)
+  private long numPutsRequested = 0, numPutsDone = 0, numPutsSkipped = 0;
+  private long lastTimeLoggedCounts = System.currentTimeMillis();
+  private long logCountsInterval = 60000;
```
Comment (jsjtzyy) [inline ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java]: make this configurable?

File: ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java [hunk 4]
```diff
@@ -115,6 +119,8 @@ public void put(MessageWriteSet messageSetToWrite) throws StoreException {
     // Write the blobs in the message set
     CloudWriteChannel cloudWriter = new CloudWriteChannel(this, messageSetToWrite.getMessageSetInfo());
     messageSetToWrite.writeTo(cloudWriter);
+
+    logPutCountsPeriodically();
```
Comment (jsjtzyy) [inline ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java]: I feel like `logPutCountsPeriodically()` depends on `put` method. That is, if `put` method, for some reason, is not invoked then `logPutCountsPeriodically()` won't be called. I am thinking if a background thread to periodically log put counts would be an option. What do you think?

File: ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java (hunk 1 above)
Comment (lightningrob) [reply ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java]: These are logged per partition, whereas the sensors are per host.  Adding sensors for each partition will become unwieldy.  This is strictly for debugging.

Comment (zzmao) [reply ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java]: ok.

File: ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java (hunk 4 above)
Comment (zzmao) [reply ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java]: That's a good point. While put rate is very small, we may not be able to see the print out, but a background thread is a little bit heavy.

Comment (jsjtzyy) [reply ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java]: Another way (maybe long-term approach) is to implement VCR's own stats manager to collect all statistics we are interested in and report to Helix or back up locally.

File: ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java (hunk 3 above)
Comment (lightningrob) [reply ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java]: I may not check in these logging changes.  I'm testing them on one EI host.

**Comment by cgtz** [comment]
@lightningrob, should this PR be closed?

**Pull request closed without merging**